docs/error-handling.md

```diff
@@ -35,4 +35,10 @@ RetryInterval: '00:10:00'
 
 ## Global Error handling
 
-The WorkflowHost service also has a `.OnStepError` event which can be used to intercept exceptions from workflow steps on a more global level.
\ No newline at end of file
+The WorkflowHost service also has a `.OnStepError` event which can be used to intercept exceptions from workflow steps on a more global level.
+
+## Custom Error Handlers 
```

Comment:
This is really an internal mechanism for handling error types, I suppose you could modify or extend it by registering other handlers in the IoC container, as you can do with any internal workflow-core service... however, I don't think this description really captures that or gives good examples on how to do it.  Maybe a better approach would be to write a doc on how to replace internal workflow-core services via IoC registration?  We could also add a reference to the new middleware feature on the error handling page?

Reply:
That's a good idea, I can open a new doc page for internal services that will, over time, contain all services documentation.
As for the middleware feature documentation - we can paste [https://github.com/danielgerlag/workflow-core/releases/tag/3.3](the detailed explanation you wrote here) into the docs